source/common/upstream/load_balancer_impl.h

```diff
@@ -27,8 +27,11 @@ class LoadBalancerBase : public LoadBalancer {
   // A utility function to chose a priority level based on a precomputed hash and
   // a priority vector in the style of per_priority_load_
   //
-  // Returns the priority, a number between 0 and per_priority_load.size()-1
-  static uint32_t choosePriority(uint64_t hash, const std::vector<uint32_t>& per_priority_load);
+  // Returns the priority, a number between 0 and per_priority_load.size()-1 as well as whether
+  // it came from the degraded priority list.
+  static std::pair<uint32_t, bool> choosePriority(uint64_t hash,
```

Comment:
Instead of a bool here, can we potentially use an enum with 2 values? I think it would increase readability a lot. Same elsewhere.